pilot/pkg/model/sidecar.go

```diff
@@ -419,11 +447,49 @@ func (sc *SidecarScope) DependsOnNamespace(namespace string) bool {
 		return true
 	}
 
-	if _, f := sc.namespaceDependencies[namespace]; f {
-		return true
+	_, exists := sc.namespaceDependencies[namespace]
+	return exists
+}
+
+// DependsOnConfig determines if the Sidecar depends on the given config.
+// Returns whether depends on this config and whether this kind of config is scoped here.
+func (sc *SidecarScope) DependsOnConfig(kind resource.GroupVersionKind, name string) (bool, bool) {
+	if sc == nil {
+		return false, false
+	}
+
+	d, exists := sc.configDependencies[kind]
```

Comment:
This looks good. But conceptually the `namespace` is not a config kind and add it to `configDependencies` may be a bit of hack. Such as using empty kind and name field to indicate that?

What's more, we haven't reach a conclusion about how to carry enough data in `PushRequest(or XdsEvent).ConfigsUpdated`. Maybe this ConfigKey struct can also be used there. Generally I'd like to defer the whole work on `ConfigsUpdated` and `configDependencies` in next PR. Do u mind that?

Reply:
What I mean is some virtual service foo in default ns we create a key (virtual service, foo, default). Every time we get an update we know all 3 of these items, right?

Reply:
OK, I misunderstood that and thought u wanted the `namespaceDependencies` field also merged into `configDependencies`.

I agree with the idea that fully-describing the config updated. As mentioned, I'd like to do it in next PR together with refactoring the `ConfigsUpdated` to obtain a total effect.

Reply:
There are two Gateway types

Reply:
I am not sure why this needs to be a part of the next PR rather than this one. Most of my concerns with this PR is the complexity of the code and how hard it is to follow which would be addressed by this change I believe?